pkg/extensions/search/schema.graphql

```diff
@@ -111,12 +92,59 @@ type OsArch {
      Arch: String
 }
 
+enum SortCriteria {
+     RELEVANCE
+     UPDATE_TIME
+     ALPHABETIC_ASC
+     ALPHABETIC_DSC
+     STARS
+     DOWNLOADS
+}
+
+# Information on current page returned by the API
+# ObjectCount represents the total number of objects on all pages
+# PreviousPage represents previous page number
+# NextPage represents next page number
+# Pages represents total number of pages
+type PageInfo {
+     ObjectCount: Int!
+     PreviousPage: Int
+     NextPage: Int
+     Pages: Int
+}
+
+# Pagination parameters
+input PageInput {
+  limit: Int
+  offset: Int
+  sortBy: SortCriteria
+}
+
+# Paginated list of RepoSummary objects
+# If limit is -1, pagination is disabled
+type PaginatedReposResult {
+     Page: PageInfo
+     Results: [RepoSummary!]!
+}
+
+# Paginated list of ImageSummary objects
+# If limit is -1, pagination is disabled
+type PaginatedImagesResult {
+     Page: PageInfo
+     Results: [ImageSummary!]!
+}
+
 type Query {
-  CVEListForImage(image: String!) :CVEResultForImage 
-  ImageListForCVE(id: String!) :[ImgResultForCVE]
-  ImageListWithCVEFixed(id: String!, image: String!) :ImgResultForFixedCVE
-  ImageListForDigest(id: String!) :[ImgResultForDigest]
-  ImageListWithLatestTag:[ImageInfo]
-  ExpandedRepoInfo(repo: String!):RepoInfo
-  GlobalSearch(query: String!): GlobalSearchResult
+     CVEListForImage(image: String!): CVEResultForImage!
+     ImageList(repo: String!, requestedPage: PageInput): PaginatedImagesResult!  # Given a repo return all contained images
+     ImageListForCVE(id: String!, requestedPage: PageInput): PaginatedImagesResult!  # Given a CVE ID retuns all affected images
+     ImageListWithCVEFixed(id: String!, image: String!, requestedPage: PageInput): PaginatedImagesResult! # Given a CVE ID retun images in which the CVE has been fixed
+     ImageListForDigest(digest: String!, requestedPage: PageInput): PaginatedImagesResult!  # Given a digest stirng return images which contain the blob with this digest
+     ImageListForLayer(digest: String!, requestedPage: PageInput): PaginatedImagesResult!  # Given a layer digest return images which contain it
+     ImageListForAscendantImage(image: String!, requestedPage: PageInput): PaginatedImagesResult!  # Given an ascendant image, identify descendent images
```

Comment:
Get images in which all of the layers of this image are included

Alternatives:
- DescendantImageList
- DerivedImageList
- DependencyListForImage
- ImageListContainingAllLayersOfGivenImage